### PR TITLE
simplify dbkvs config

### PR DESCRIPTION
**Pull request by rjullman**
Took a pass at simplifying all the configuration setup.

### CODE REVIEW

**Comment by rjullman** [comment]
Ended up having to add in h2 in memory config options to get internal tests to work with this stricter config setup.  Sorry to complicate the PR.  If it is too crazy I can try to split this up.

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/H2DdlConfig.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.dbkvs;
```
Comment (jboreiko) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/H2DdlConfig.java]: nit: I wouldd

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DdlConfig.java [hunk 2]
```diff
@@ -15,17 +15,23 @@
...
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
-import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.google.common.base.Supplier;
 import com.palantir.atlasdb.AtlasDbConstants;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.DbTableFactory;
+
+@JsonTypeInfo(use = Id.NAME, include = As.PROPERTY, property = "type", visible = false)
+public abstract class DdlConfig {
```
Comment (jboreiko) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DdlConfig.java]: What value do each of the instantiation options provide?  Seems like unless we are going to override them with defaults per kvs type they aren't necessary.

**Comment by jboreiko** [comment]
Did not intend to close this issue.

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/OracleConnectionConfig.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    public abstract Optional<String> getServerDn();
+
+    @Value.Default
+    public boolean getMatchServerDn() {
+        return getServerDn().isPresent();
+    }
+
+    @Value.Default
+    public boolean getTwoWaySsl() {
+        return false;
```
Comment (tpetracca) [inline atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/OracleConnectionConfig.java]: So I know this used to be false, but given the infosec push I'm pretty sure we should be setting it to true, even if we think that's breaking.

File: atlasdb-ete-test-utils/src/main/java/com/palantir/atlasdb/ete/Gradle.java [hunk 4]
```diff
@@ -42,10 +42,10 @@ public static Gradle ensureTaskHasRun(String command) {
     @Override
     protected void before() throws Throwable {
         if (isRunningOutsideOfGradle()) {
-            System.out.println("It looks like you are not running in gradle, performing the required gradle command: " + command);
+            System.out.println("It looks like you are not running in gradle, performing the required gradle command: " + command);  // (authorized)
```
Comment (tpetracca) [inline atlasdb-ete-test-utils/src/main/java/com/palantir/atlasdb/ete/Gradle.java]: out of curioustiy, what is that comment supposed to mean / do?

Comment (jboreiko) [inline atlasdb-ete-test-utils/src/main/java/com/palantir/atlasdb/ete/Gradle.java]: This is a utility class that helps determine whether or not a necessary gradle task has been run to kick off testing.  It allows you to run the test in intelij but use the same dependent gradle tasks that can't be easily replicated in intelij @joelea who set this up

Comment (tpetracca) [inline atlasdb-ete-test-utils/src/main/java/com/palantir/atlasdb/ete/Gradle.java]: I was vague there.  I was specifically asking about the purpose of the "// (authorized)".

Comment (jboreiko) [inline atlasdb-ete-test-utils/src/main/java/com/palantir/atlasdb/ete/Gradle.java]: this is super old, but was in my queue, I believe this comes from a checkstyle we have (at least internally) that prevents any system.out.printlns that don't have an authorized block.

**Pull request merged**